pkg/discovery/discovery.go

```diff
@@ -97,6 +98,9 @@ func (td *tidbDiscovery) Discover(advertisePeerUrl string) (string, error) {
 
 	if len(currentCluster.peers) == int(replicas) {
 		delete(currentCluster.peers, podName)
+		if len(pdAddresses) != 0 {
+			return fmt.Sprintf("--join=%s", strings.Join(pdAddresses, ",")), nil
+		}
```

Comment:
I suggest visiting PD API first and querying the latest member address as PD's count may scale-in or scale-out and the information here would be out-of-date.

Reply:
This argument will only be used by one of PD Pod during the new tidb cluster creation. After the Pod join the exsiting cluster successfully, the others will get PD members by PD API. If the `PDAdresses` is not correct, **the Pod** is not able to join the exsiting cluster, thus PD API won't return any value to the other PD Pods. And the Pod used incorrect `PDAddresses` will restart soon, until the `PDAddresses` is configured correctly.